runtime/jcl/common/vectornatives.cpp

```diff
@@ -70,4 +70,29 @@ Java_jdk_internal_vm_vector_VectorSupport_getMaxLaneCount(JNIEnv *env, jclass cl
 	return laneCount;
 }
 
+#if JAVA_SPEC_VERSION >= 25
+jstring JNICALL
+Java_jdk_internal_vm_vector_VectorSupport_getCPUFeatures(JNIEnv *env, jclass clazz)
+{
+	char buf[1024];
+	OMRPORT_ACCESS_FROM_J9VMTHREAD((J9VMThread *)env);
+	OMRProcessorDesc processorDesc;
+	if (0 == omrsysinfo_get_processor_description(&processorDesc)) {
+		if (0 != omrsysinfo_get_processor_feature_string(&processorDesc, buf, sizeof(buf))) {
+			PORT_ACCESS_FROM_ENV(env);
+			int biggerSize = 8192;
+			jstring result = NULL;
+			char *newbuf = (char *)j9mem_allocate_memory(biggerSize, J9MEM_CATEGORY_VM_JCL);
+			if (NULL != newbuf) {
```

Comment:
Like you suggested in https://github.com/eclipse-openj9/openj9/pull/22253#discussion_r2216864506, this should probably just use a fixed-size (1024) buffer.